components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/GleanLifecycleObserver.kt

```diff
@@ -40,5 +40,14 @@ internal class GleanLifecycleObserver : LifecycleObserver {
         // because it belongs to the baseline ping and that ping is sent every
         // time the app goes to background.
         GleanBaseline.duration.start()
+
+        // We check for the metrics ping schedule here because the app could have been in the
+        // background and resumed in which case Glean would already be initialized but we still need
+        // to perform the check to determine whether or not to collect and schedule the metrics ping.
+        // If this is the first ON_START event since the app was launched, Glean wouldn't be
+        // initialized yet.
+        if (Glean.isInitialized()) {
+            Glean.metricsPingScheduler.startupCheck()
```

Comment:
The risk of this change might be that when it enters foreground multiple times during the same 24 hour period, the ping is scheduled for sending twice (at the same 4AM instant the next morning).  I think the fact that we use `WorkManager.getInstance().enqueueUniqueWork()` makes that impossible, but I wonder if there's a way to write a test for that.

Reply:
I could also check to see if a `Worker` is already enqueued, which I didn't really think to do either.  I agree we should be okay since we do use the `enqueueUniqueWork` but a test and/or a check wouldn't hurt.

Reply:
One test could be to use the workmanager testing api to enqueue two unique work units and check if both are there. But that would only test the WorkManager API, and I bet that such a test is already there, for them